### PR TITLE
[T3A4][T13-A2] Kevin Ong

### PR DESCRIPTION
**Pull request by Kevin-Ong**
Ready for review

### CODE REVIEW

File: src/seedu/addressbook/storage/StorageFile.java [hunk 1]
```diff
@@ -88,9 +88,20 @@ public void save(AddressBook addressBook) throws StorageOperationException {
...
          */
-        try (final Writer fileWriter =
-                     new BufferedWriter(new FileWriter(path.toFile()))) {
+        try {
+            if (!path.toFile().exists()) {
+                throw new FileNotFoundException();
+            }
+            writeFile(addressBook);
+        } catch (FileNotFoundException fnfe) {
+            System.out.println("Storage file deleted! Recreating...");
```
Comment (okkhoy) [inline src/seedu/addressbook/storage/StorageFile.java]: what happens to SLAP when you print here? think!

there is already a class that performs user interaction, think how and where you can use that to show the error message?

i like this approach :)

File: src/seedu/addressbook/storage/StorageFile.java [hunk 2]
```diff
@@ -88,9 +88,20 @@ public void save(AddressBook addressBook) throws StorageOperationException {
...
+            writeFile(addressBook);
```
Comment (okkhoy) [inline src/seedu/addressbook/storage/StorageFile.java]: think if there is a way to eliminate this (sort of) duplicate call in the same try-catch block,

**Comment by okkhoy** [comment]
@Kevin-Ong i liked your approach! 
Some comments, to help you think through to refine the solution, added. Please ack & close the PR after reading comments.

**Comment by Kevin-Ong** [comment]
Acknowledged :)

**Pull request closed without merging**